src-tauri/src/compressor.rs

```diff
@@ -1,12 +1,141 @@
-use bitvec::prelude::BitVec;
+use lz4::{Decoder, EncoderBuilder};
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader, Result, Write},
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
 
+// Trait for all compression and decompression functions
 pub trait Compressor {
```

Comment:
also can you edit main.rs to add the compression during encode_sequence? 

https://github.com/UBC-iGEM/dna-software/blob/151916379ad340278c7996866aa9d992d4827303/src-tauri/src/main.rs#L35-L47


it should follow the old deleted sequence.rs almost exactly

https://github.com/UBC-iGEM/dna-software/blob/2b88993ed61a17033ea4fb30c28932cfbc92ae6f/src-tauri/src/sequence.rs#L11-L21